### PR TITLE
Make render8 pages work for stores running on portal

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What does this PR do? \*

- ditto.

#### How to test it? \*

- https://feirissima.vtexcommercestable.com.br/_secure/account?workspace=felipe#/subscriptions
- visit the Subscriptions on my-account page

### CODE REVIEW

File: react/utils/host.ts [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
+  const rootDoc = getRootDocument()
+  const generatorMetaTag = rootDoc.querySelector(`meta[name='generator']`)
   const generator = generatorMetaTag && generatorMetaTag.getAttribute('content')
-  return generator && generator.startsWith('vtex.render-server')
+
+  const isRender8 = generator && generator.startsWith('vtex.render-server')
+
+  const isServerdThroughJanus = window.location.pathname.startsWith('/api/io')
+
+  return isServerdThroughJanus ? false : !!isRender8
```
Comment (Klynger) [inline react/utils/host.ts:20]: Typo
```suggestion
  return isServedThroughJanus ? false : Boolean(isRender8)
```

File: react/utils/host.ts [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+  return isIframe() ? window.top.document : document
+}
+
+function isRenderServedPage(): boolean {
+  const rootDoc = getRootDocument()
+  const generatorMetaTag = rootDoc.querySelector(`meta[name='generator']`)
   const generator = generatorMetaTag && generatorMetaTag.getAttribute('content')
-  return generator && generator.startsWith('vtex.render-server')
+
+  const isRender8 = generator && generator.startsWith('vtex.render-server')
```
Comment (Klynger) [inline react/utils/host.ts:16]:
```suggestion
  const isRender8 = generator?.startsWith('vtex.render-server')
```

File: react/utils/host.ts [hunk 3]
```diff
@@ -1,19 +1,34 @@
...
+
+  const isServerdThroughJanus = window.location.pathname.startsWith('/api/io')
```
Comment (Klynger) [inline react/utils/host.ts:18]: Typo
```suggestion
  const isServedThroughJanus = window.location.pathname.startsWith('/api/io')
```

File: react/utils/host.ts (hunk 1 above)
Comment (Klynger) [inline react/utils/host.ts:20]:
```suggestion
  return !isServedThroughJanus && Boolean(isRender8)
```

**Review by Klynger — CHANGES_REQUESTED** [review]

**Comment by Klynger** [comment]
I didn't understand how to test it and when I open the URL I just see this:

image

**Comment by salesfelipe** [comment]
> I didn't understand how to test it and when I open the URL I just see this:
> 
> image

You have to login to access the my-account then go to the subscriptions section.

**Review by Klynger — APPROVED** [review]

**Review by kevinch — APPROVED** [review]
Looks good but i'm not the best expert for the render-runtime.

**Review by brunoabreu — APPROVED** [review]

**Comment by lbebber** [comment]
Screen Shot 2020-11-17 at 17 27 39
to be sure, is this expected? (i.e. the lack of proper messages)

**Comment by salesfelipe** [comment]
> Screen Shot 2020-11-17 at 17 27 39
> to be sure, is this expected? (i.e. the lack of proper messages)

Yes, I had to fix this on the my-subscriptions app, the problem is the `store/` prefix...

**Review by lbebber — APPROVED** [review]
LGTM just be extra careful with the release 🙏

**Comment by salesfelipe** [comment]
> LGTM just be extra careful with the release 🙏

:harold-pain:

**Pull request merged**